src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -467,6 +467,26 @@ def get_load_balancer_backend_pool_type(self) -> str:
         # this parameter does not need validation
         return load_balancer_backend_pool_type
 
+    def get_nrg_restriction(self) -> Union[str, None]:
+        """Obtain the value of nrg_restriction.
+        :return: string or None
+        """
+        # read the original value passed by the command
+        nrg_restriction = self.raw_param.get("nrg_restriction")
+
+        # In create mode, try to read the property value corresponding to the parameter from the `mc` object.
+        if self.decorator_mode == DecoratorMode.CREATE:
```

Comment:
@FumingZhang, I'm curious if you know the reasoning for this line.

I found it in `get_auto_upgrade_channel` method:
https://github.com/Azure/azure-cli/blob/faccf0b590f42b87163c393b7cdfc25bd73640d0/src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py#L3839

Reply:
We have two code paths, one for create and the other for update. In both of them, the parameter value is obtained through the same `get_xxx` function, but the behavior of obtaining the value is expected to be different.

In the create code path, we want to obtain the option value from the target mc first (when creating the resource, if the option value has been written into the target mc, it's considered as immutable), if it's not set yet, we'll obtain the value from user input.

In the update code path, we just obtain the value from user input.

Many options do not support being updated (can only be set at create time), so for their getter function, it doesn't matter whether to add this line (`if self.decorator_mode == DecoratorMode.CREATE:`) or not.

Reply:
Is there an example case where its expected to possibly set the value on the mc before obtaining it? Possibly an interaction between a profile being set both in the core cli + the preview cli?

Reply:
When getting the parameter value for the very first time, the corresponding value in mc shouldn't be set yet (there are cases where SDK models provide default values, but that's actually handled by some other hack). However, some cross-checks between profiles will obtain the property value ​​in the previously set profile again, and at this time the value will be read from mc (instead of user's raw input). One such example would be the validation between dns prefix and fqdn subdomain.

Reply:
Ok, interesting.